pkg/cli/install/steps.go

```diff
@@ -0,0 +1,191 @@
+// everest
+// Copyright (C) 2023 Percona LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package install
+
+import (
+	"context"
+	"fmt"
+	"path"
+
+	"github.com/AlekSi/pointer"
+	"github.com/cenkalti/backoff/v4"
+	olmv1alpha1 "github.com/operator-framework/api/pkg/operators/v1alpha1"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	"github.com/percona/everest/pkg/cli/helm"
+	helmutils "github.com/percona/everest/pkg/cli/helm/utils"
+	"github.com/percona/everest/pkg/cli/steps"
+	"github.com/percona/everest/pkg/common"
+	"github.com/percona/everest/pkg/kubernetes"
+)
+
+func (o *Install) newStepInstallEverestHelmChart() steps.Step {
+	return steps.Step{
+		Desc: "Installing Everest Helm chart",
+		F: func(ctx context.Context) error {
+			return o.installEverestHelmChart(ctx)
+		},
+	}
+}
+
+func (o *Install) newStepEnsureEverestOperator() steps.Step {
+	return steps.Step{
+		Desc: "Ensuring Everest operator deployment is ready",
+		F: func(ctx context.Context) error {
+			return o.waitForDeployment(ctx, common.PerconaEverestOperatorDeploymentName, common.SystemNamespace)
+		},
+	}
+}
+
+func (o *Install) newStepEnsureEverestAPI() steps.Step {
+	return steps.Step{
+		Desc: "Ensuring Everest API deployment is ready",
+		F: func(ctx context.Context) error {
+			return o.waitForDeployment(ctx, common.PerconaEverestDeploymentName, common.SystemNamespace)
+		},
+	}
+}
+
+func (o *Install) newStepEnsureEverestOLM() steps.Step {
+	return steps.Step{
+		Desc: "Ensuring OLM components are ready",
+		F: func(ctx context.Context) error {
+			depls, err := o.kubeClient.ListDeployments(ctx, kubernetes.OLMNamespace)
+			if err != nil {
+				return err
+			}
+			for _, depl := range depls.Items {
+				if err := o.kubeClient.WaitForRollout(ctx, depl.GetName(), depl.GetNamespace()); err != nil {
+					return err
+				}
```

Comment:
Would the logging provided by the `waitForDeployment` wrapper be useful here or would it be too much?

Reply:
Hmm.. I think it could be useful. Updated